applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <caf/events/ble_common_event.h>
+
+#if CONFIG_DESKTOP_USB_ENABLE
+#include "usb_event.h"
+#endif
+
+#define MODULE dvfs
+#include <caf/events/module_state_event.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_DVFS_LOG_LEVEL);
+
+#include "ld_dvfs_handler.h"
+
+#define DVFS_RETRY_TIMEOUT		K_MSEC(500)
+/* 0x0d00 is a magic number which codes LLPM connection */
+#define REG_CONN_INTERVAL_LLPM_MASK	0x0d00
+
+enum dvfs_state {
+	DVFS_STATE_INITIALIZING,
+	DVFS_STATE_LLPM_CONNECTED,
+	DVFS_STATE_USB_CONNECTED,
+	DVFS_STATE_COUNT
+};
+
+/* Table containing the frequency settings for each state.
+ * It is tailored for nrf54h20dk and might need to be adjusted for other platforms.
+ */
+enum dvfs_frequency_setting dvfs_freq_table[DVFS_STATE_COUNT] = {
+	DVFS_FREQ_HIGH,		/* DVFS_STATE_INITIALIZING */
+	DVFS_FREQ_MEDLOW,	/* DVFS_STATE_LLPM_CONNECTED */
+	DVFS_FREQ_HIGH		/* DVFS_STATE_USB_CONNECTED */
+	};
```

Comment:
Instead of comments:
```suggestion
enum dvfs_frequency_setting dvfs_freq_table[DVFS_STATE_COUNT] = {
	[DVFS_STATE_INITIALIZING] = DVFS_FREQ_HIGH,	
	[DVFS_STATE_LLPM_CONNECTED] = DVFS_FREQ_MEDLOW,
	[DVFS_STATE_USB_CONNECTED] = DVFS_FREQ_HIGH,
};
```